plugins/inventory/nb_inventory.py

```diff
@@ -704,6 +717,29 @@ def extract_regions(self, host):
             object_parent_lookup=self.regions_parent_lookup,
         )
 
+    def extract_location(self, host):
+        # A host may have a location. A location may have a parent location.
+        # Produce a list of locations:
+        # - it will be empty if the device has no location
+        # - it will have 1 element if the device's location has no parent
+        # - it will have multiple elements if the location has a parent location
+
+        location = host.get("location", None)
+        if not isinstance(location, dict):
+            # Device has no location
+            return []
+
+        location_id = location.get("id", None)
+        if location_id is None:
+            # Device has no location
+            return []
```

Comment:
I think this can be simplified. `location` is `None` if not set so can just return the empty list if it's `None`.

Reply:
I just copied the `extract_regions()` function, so I guess I didn't look at it to closely. Are you saying if `host['location']` is not `None` it will always have an `'id'` key set so we don't need to check it?

I can change to something like this to cover both still:
```python
location_id = host.get("location", {}).get("id", None)
if not location_id:
    # Device has no location
    return []
```
Or are you thinking just this and not bother with `'id'`:
```python
if not host.get("location", None):
    # Device has no location
    return []
location_id = host["location"]["id"]
```

Reply:
Correct. I think that should be the case.

Reply:
I went with
```python
try:
    location_id = host["location"]["id"]
except (KeyError, TypeError):
    # Device has no location
    return []
```
which is probably more pythonic...